### DISCUSSION TITLE
How to achieve the Growing Index in 2.3.0

### DISCUSSION THREAD

**Discussion by GitHuY** [milvus-io/milvus#26781] [Q&A and General discussion]
Hi, team.
I find the 2.3.0 Release Note that it introduce the Growing index, which automatically creates real-time indexes for streaming data to improve query performance. But the index building is very time-consuming, could you give a brief introduction how Milvus achieve it in real-time?

**Comment by xiaofan-luan** [comment]
growing index should be automatically enabled on streaming data

**Comment by xiaofan-luan** [comment]
Milvus will wait until 10% of the growing segment inserted.
Then we do cluster and build IVF index. the rest of the data will be assigned to one of the cluster in based on their distance to centroids.
When search, it's similar to IVF index

**Reply by GitHuY** [reply]
Thanks for your answer, xiaofan. Two follow-up questions:
1. It means before the data increased to 10%, Milvus will still do the brute-force search on them ?
2. It the collection's original index is HNSW, the growing index is also IVF?

**Comment by xiaofan-luan** [comment]
> Thanks for your answer, xiaofan. Two follow-up questions:
> 
> 1. It means before the data increased to 10%, Milvus will still do the brute-force search on them ?
> 2. It the collection's original index is HNSW, the growing index is also IVF?

1. yes, but it's gonna to much faster because the data size is small
2. yes,  growing index is not controlled by and index build and search params.
